docs/compiling-mono/windows/index.md

```diff
@@ -4,51 +4,104 @@ redirect_from:
-cd mono-$VERSION
-./configure --prefix=$PREFIX --host=i686-w64-mingw32
+Below we specify 64-bit build and sgen GC (The boehm GC is not supported on 64-bit windows), installing build into `$PREFIX` where mono distribution already exists (replacing). NOTE, in order to keep existing mono distribution untouched, setup a different 
+`$PREFIX`. You may also only set `PATH` to point to existing mono distribution and not specify `$PREFIX` at all.
+
+```
+PATH=$PREFIX/bin:$PATH
+git clone https://github.com/mono/mono.git
+cd mono
+./autogen.sh --prefix=$PREFIX --host=x86_64-w64-mingw32 --disable-boehm
+```
+
+Now open the mono solution `msvc\mono.sln` in Visual Studio. (Re)Build the solution and point the `MONO_EXECUTABLE` to the mono exe you just built in Visual Studio. Eg:
+```
+export MONO_EXECUTABLE=./msvc/build/sgen/x64/bin/Release/mono-sgen.exe 
+```
+if you built 64-bit Debug version.
+
+Next step is to build the BCL and all System libraries. Go back to cygwin and run make:
+```
 make
 make install
 ```
 
-By the end of this process, you will have Mono installed on your PREFIX directory.
+It is possible to parallelize the cygwin build passing in for example `-j8`,  where 8 in this case is the number of parallel builders to run.
+
+To build Mono 32-bit, run autogen with `--host=i686-w64-mingw32` instead and choose 32-bit when building in Visual Studio.
+
+### Command line using msbuild.exe
+Instead of building from within Visual Studio you can use `msbuild.exe` directly in the cygwin terminal:
+```
+/cygdrive/c/Program\ Files\ \(x86\)/MSBuild/14.0/Bin/MSBuild.exe /p:PlatformToolset=v140 /p:Platform=x64 /p:Configuration=Release /p:MONO_TARGET_GC=sgen msvc/mono.sln
+```
+Use desired parameters in the Platform and Configuration parameters to change between 64 and 32 bit builds as well as Release and Debug builds.
 
-Building Mono From a Git Source Code Checkout
----------------------------------------------
+If only changes has been done to the native code and no changes have been 
+done to base class libraries or .NET tests, building/rebuilding using Visual Studio solution 
+is enough. If, however there are changes done to the base class libraries or tests, you need to run make in cygwin to rebuild the class libraries and/or tests.
 
-To build Mono from a Git Source Code checkout, you will want to have the official Mono installed on the system and in your PATH, as the build requires a working C# compiler to run.
+## Build with no Mono distribution available
 
-On Windows, you should set the following Git setting to avoid issues with line endings: `git config --global core.autocrlf input`
+### Prerequisites:
+* 64-bit Cygwin.
+* All packages as listed above.
+* Visual Studio 2015 (if custom branch is used).
 
-Once you do this, run the following commands, remember to replace PREFIX with your installation prefix that you selected:
+### Steps
 
-``` bash
-PATH=$PREFIX/bin:$PATH
+* Open cygwin prompt.
+```
+git config --global core.autocrlf input
```

Comment:
If we mention this here we should mention it in the earlier section where we do `git clone` as well.